src/main/java/org/broadinstitute/barclay/argparser/CommandLineException.java

```diff
@@ -90,16 +91,17 @@ public CommandLineParserInternalException( final String s, final Throwable throw
 
     /**
      * For wrapping errors that are believed to never be reachable
+     * Package protected to restrict usage to the arg parsers.
      */
-    public static class ShouldNeverReachHereException extends CommandLineException {
+    static class ShouldNeverReachHereException extends RuntimeException {
```

Comment:
Should probably extend `CommandLineParserInternalException`, which should be the superclass of all internal errors.